### PR TITLE
Add `seq-hasher`

### PR DESCRIPTION
**Pull request by apcamargo**
This PR adds [`seq-hasher`](https://github.com/apcamargo/seq-hasher), a tool for fast computation of hash digests for DNA sequences in FASTA files, with support for circular permutations.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by apcamargo** [comment]
@BiocondaBot please add label

File: recipes/seq-hasher/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -euo
+
+set -xe
+
+# build statically linked binary with Rust
+RUST_BACKTRACE=1 cargo install --verbose --root $PREFIX --path .
```
Comment (mbhall88) [inline recipes/seq-hasher/build.sh:6]:
```suggestion
RUST_BACKTRACE=1 cargo install --verbose --locked --root $PREFIX --path .
```

**Review by mbhall88 — CHANGES_REQUESTED** [review]

**Comment by mbhall88** [comment]
I would also suggest following the conda-forge documentation for bundling licenses from dependencies. https://conda-forge.org/docs/maintainer/adding_pkgs/#rust

File: recipes/seq-hasher/meta.yaml [hunk 2]
```diff
@@ -6,16 +6,19 @@ package:
 
 build:
   number: 0
+  script:
+    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
```
Comment (mbhall88) [inline recipes/seq-hasher/meta.yaml:10]: as you have a `build.sh`, adding this `script:` directive will make the build not use that script I believe. Move the `cargo-bundle-licenses --format yaml --output THIRDPARTY.yml` into the `build.sh`. Take a look at https://github.com/conda-forge/ssubmit-feedstock/tree/main/recipe for an example

**Review by mbhall88 — CHANGES_REQUESTED** [review]

File: recipes/seq-hasher/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  run:
+
+test:
+  commands:
+    - seq-hasher --help
+
+about:
+  home: https://github.com/apcamargo/seq-hasher
+  license: MIT
+  license_file: LICENSE
```
Comment (mbhall88) [inline recipes/seq-hasher/meta.yaml:30]:
```suggestion
  license_file: 
    - LICENSE
    - THIRDPARTY.yml
```

**Review by mbhall88 — CHANGES_REQUESTED** [review]

**Comment by mbhall88** [comment]
And while I'm being annoying, I might also recommend adding `--no-track` to the install command https://github.com/conda-forge/conda-forge.github.io/issues/2191

File: recipes/seq-hasher/meta.yaml [hunk 4]
```diff
@@ -27,7 +27,10 @@ test:
 about:
   home: https://github.com/apcamargo/seq-hasher
   license: MIT
-  license_file: LICENSE
+  license_file:
+  - LICENSE
+  - THIRDPARTY.yml
```
Comment (mbhall88) [inline recipes/seq-hasher/meta.yaml:32]: I believe these will need to be indented

**Review by mbhall88 — APPROVED** [review]

**Pull request merged**